Crazy DNS packet:

$ dig @64.29.144.70 www.bookride.com

; <<>> DiG 9.3.6-P1-RedHat-9.3.6-4.P1.el5_4.2 <<>> @64.29.144.70 www.bookride.com
; (1 server found)
;; global options:  printcmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 10397
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 1, ADDITIONAL: 0

;; QUESTION SECTION:
;www.bookride.com.              IN      A

;; ANSWER SECTION:
www.bookride.com.       3600    IN      CNAME   ghs.google.com.

;; AUTHORITY SECTION:
google.com.             86400   IN      SOA     ns1.aplus.net. hostmaster.aplus.net. 1007 86403 3600 3600000 86400

;; Query time: 304 msec
;; SERVER: 64.29.144.70#53(64.29.144.70)
;; WHEN: Sat Aug 14 01:18:15 2010
;; MSG SIZE  rcvd: 119

$ dig @64.29.154.70 www.bookride.com

; <<>> DiG 9.3.6-P1-RedHat-9.3.6-4.P1.el5_4.2 <<>> @64.29.154.70 www.bookride.com
; (1 server found)
;; global options:  printcmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 63076
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 1, ADDITIONAL: 0

;; QUESTION SECTION:
;www.bookride.com.              IN      A

;; ANSWER SECTION:
www.bookride.com.       3600    IN      CNAME   ghs.google.com.

;; AUTHORITY SECTION:
google.com.             86400   IN      SOA     ns1.aplus.net. hostmaster.aplus.net. 1007 86403 3600 3600000 86400

;; Query time: 132 msec
;; SERVER: 64.29.154.70#53(64.29.154.70)
;; WHEN: Sat Aug 14 01:18:26 2010
;; MSG SIZE  rcvd: 119

This patch allows Deadwood to handle insanity like this.

--- deadwood-2.9.04/src/DwRecurse.c	2010-08-13 01:48:01.000000000 -0700
+++ deadwood-2.9.05/src/DwRecurse.c	2010-08-14 01:10:01.000000000 -0700
@@ -1495,9 +1495,11 @@
 
         /* If the NXDOMAIN bit is set in the header, the answer must look like
          * a NXDOMAIN */
-        if(in_type == TYPE_NXDOMAIN && type != TYPE_NXDOMAIN) {
+	/* Disabled: bookride.com screws this up, so there are out there in
+         * the wild answers marked NXDOMAIN that actually aren't */
+        /*if(in_type == TYPE_NXDOMAIN && type != TYPE_NXDOMAIN) {
                 return 0;
-        }
+        }*/
 
         if(type == TYPE_CNAME_REFER) {
                 out = dwx_make_cname_refer(view);
